packages/components/tests/dummy/app/templates/components/form/base-elements.hbs

```diff
@@ -0,0 +1,355 @@
+    structured content to it (it's just yielded in output):
+  </p>
+  {{! prettier-ignore-start }}
+  <CodeBlock
+    @language="markup"
+    @code='
+      <Hds::Form::Label @controlId="control-ID">
+        <span>Some text</span>
+        <Hds::Badge @size="small" @text="Some badge" />
+      </Hds::Form::Label>
+    '
+  />
+  {{! prettier-ignore-end }}
+  <p class="dummy-paragraph">Renders to:</p>
+  <Hds::Form::Label @controlId="control-ID">
+    <span>Some text</span>
+    <Hds::Badge @size="small" @text="Some badge" />
+  </Hds::Form::Label>
+  <p class="dummy-paragraph">
+    <strong>Important:</strong>
+    in this case, while the correct text styling is applied to the component's container, the layout/organization of the
+    content inside the component is left to the consumer.</p>
+  <p class="dummy-banner dummy-banner--info dummy-paragraph"><span aria-hidden="true">⚠️</span>
+    The
+    <code class="dummy-code">&lt;label&gt;</code>
+    element is linked via
+    <code class="dummy-code">for</code>
+    attribute to the
+    <code class="dummy-code">&lt;input/select/textarea&gt;</code>
+    elements. This means it becomes an interactive element, and for this reasons it's not possible to have links inside
+    it (nested interactive elements are not accessible).
+  </p>
+
+  <h4 class="dummy-h4">Form::HelperText</h4>
+  <p class="dummy-paragraph">
+    The most basic invocation just needs a text passed to the component and a
+    <code class="dummy-code">controlId</code>
+    argument:
+  </p>
+  {{! prettier-ignore-start }}
+  <CodeBlock
+    @language="markup"
+    @code='
+      <Hds::Form::HelperText @controlId="control-ID">This is some helper text</Hds::Form::HelperText>
+    '
+  />
+  {{! prettier-ignore-end }}
+  <p class="dummy-paragraph">Renders to:</p>
+  <Hds::Form::HelperText @controlId="control-ID">This is some helper text</Hds::Form::HelperText>
+  <p class="dummy-paragraph"><em>Notice: the
+      <code class="dummy-code">controlId</code>
+      value will be used to generate an ID, prefixed with
+      <code class="dummy-code">helper-text-</code>, so that this ID can be referenced in the
+      <code class="dummy-code">aria-describedby</code>
+      attribute of the form control. If no
+      <code class="dummy-code">controlId</code>
+      is provided, no ID is generated (but if needed it can be passed directly as HTML attribute).</em></p>
+  <p class="dummy-paragraph">
+    There may be cases in which the helper text needs to contain more than just text. In this case it's possible to pass
+    structured content to it (it's just yielded in output):
+  </p>
+  {{! prettier-ignore-start }}
+  <CodeBlock
+    @language="markup"
+    @code='
+      <Hds::Form::HelperText @controlId="control-ID">
+        Some text with a
+        <Hds::Link::Inline @route="components.link.inline">Hds::Link::Inline</Hds::Link::Inline>,
```

Comment:
did you get a chance to review https://codepen.io/melsumner/pen/NWyLMOQ? We may need to rethink the docs here.

Reply:
@MelSumner this refers to the `helper text`, which is a normal element, not a `<label>` (I have updated the documentation removing any reference to link in the label, and added a warning)
<img width="1070" alt="screenshot_1544" src="https://user-images.githubusercontent.com/686239/172928829-ae011b9a-fcb2-4f98-acb3-a644c0a998b6.png">

Reply:
I suppose this will fall into the category of "technically permissible but doesn't work" - anything that is referred to in `aria-describedby` has only the text read out, so the user won't know that it's actually a link.

Let's brainstorm with the team and see if there's an elegant way to solve this problem.

Reply:
@MelSumner will
- update the documentation for the `HelperTex` in the `base-elements` page, with a note about how to make the content more accessible if there is a link inside the helper text
- create a helper component `ScreenReaderOnly` (name tbd) that can wrap a generic content/text with the existing `.sr-only` class, to make the content available only to screen readers.